bionic/deriver.py

```diff
@@ -94,18 +94,17 @@ def should_include_entity_name(name):
                     tasks, key=lambda task: task.keys[0].case_key)):
                 task_key = task.key_for_entity_name(entity_name)
                 state = self._task_states_by_key[task_key]
-
                 node_name = name_template.format(
                     entity_name=entity_name, task_ix=task_ix)
-
                 graph.add_node(
                     task_key,
                     name=node_name,
                     entity_name=entity_name,
                     case_key=task_key.case_key,
                     task_ix=task_ix,
+                    # TODO: Get from provider
+                    docstring=self._docstring_by_entity_name.get(entity_name) or "",
```

Comment:
Two small nits here:

1. For consistency, let's reserve `docstring` for actual Python docstrings and use `doc` when describing entity documentation (which might or might not come from an actual docstring). This isn't obvious from the code but it's a pattern I tried to follow when I updated the docstring stuff.

2. It doesn't make much difference here, but as a general rule I'd avoid replacing `None` with `""` until the point where it's actually necessary. In this particular case, you have an attribute called `docstring`, and the entity doesn't actually have a docstring, so the most appropriate value here would seem to be `None`. When we get to the point of converting the docstring into a tooltip, then we may want to substitute in placeholder values like `""`. (Although -- is `""` better than not having a tooltip at all?)

Reply:
I moved the string handling to dagviz.py when we actually get the docs for entities- also wrote a test for it. If we want to do any additional string cleaning it could happen here too